src/components/header/Header.js

```diff
@@ -0,0 +1,91 @@
+import { useState, useEffect, useLayoutEffect, useRef } from 'react';
+
+// Hamburger Menu Icons
+import { MdClose } from 'react-icons/md';
+import { FiMenu } from 'react-icons/fi';
+
+import { motion } from 'framer-motion';
+import useWindowSize from 'hooks/useWindowSize';
+import NavMenu from './NavMenu';
+
+export function Header() {
+    // If theme state is false, app is on dark mode.
+    const [theme, setTheme] = useState(false);
+    const [navbarOpen, setNavbarOpen] = useState(false);
+    const headerContainer = useRef(null);
+    const logoSurname = useRef(null);
+
+    // Get window size with useWindowSize custom hook
+    const windowSize = useWindowSize();
+    const isMobile = windowSize.width < 768;
+
+    const handleToggle = () => {
+        setNavbarOpen((prevState) => !prevState);
+    };
+
+    const handleThemeToggle = () => {
+        setTheme((prevState) => !prevState);
+    };
+
+    useEffect(() => {
+        document.body.classList.toggle('light');
+    }, [theme]);
+
+    useLayoutEffect(() => {
+        const handleScroll = () => {
+            const isScrolling = window.scrollY > 0;
+            if (!isMobile && isScrolling) {
+                headerContainer.current.classList.add('scrolling');
+                logoSurname.current.classList.add('logo-surname');
+            }
+            if (!isScrolling) {
+                headerContainer.current.classList.remove('scrolling');
+                logoSurname.current.classList.remove('logo-surname');
+            }
+        };
+        window.addEventListener('scroll', handleScroll);
+
+        return () => {
+            window.removeEventListener('scroll', handleScroll);
+        };
+    });
```

Comment:
As we only need to set the scroll event listener once, should we make this only run on mount?

Reply:
Fixed it.